src/main/java/com/greghaskins/spectrum/internal/configuration/BlockTimeout.java

```diff
@@ -0,0 +1,76 @@
+package com.greghaskins.spectrum.internal.configuration;
+
+import static com.greghaskins.spectrum.internal.hooks.Hook.withUnreportedInnerBlock;
+
+import com.greghaskins.spectrum.Block;
+import com.greghaskins.spectrum.internal.Child;
+import com.greghaskins.spectrum.internal.LeafChild;
+import com.greghaskins.spectrum.internal.hooks.Hook;
+
+import org.junit.internal.runners.statements.FailOnTimeout;
+import org.junit.runners.model.Statement;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Applies timeout metadata to the block. A timeout will cause a leaf child to
+ * run within a sentinel which fails the test if it takes too long.
+ */
+public class BlockTimeout implements BlockConfigurable<BlockTimeout> {
+  private long timeout;
+  private TimeUnit timeUnit;
+
+  /**
+   * Create a timeout in milliseconds.
+   * @param timeout millisecond timeout
+   */
+  public BlockTimeout(long timeout) {
+    this(timeout, TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Create a timeout in any {@link TimeUnit}.
+   * @param timeout timeout amount
+   * @param timeUnit time unit
+   */
+  public BlockTimeout(long timeout, TimeUnit timeUnit) {
+    this.timeout = timeout;
+    this.timeUnit = timeUnit;
+  }
+
+  @Override
+  public boolean inheritedByChild() {
+    return true;
+  }
+
+  @Override
+  public void applyTo(Child child, TaggingFilterCriteria state) {
+    if (child instanceof LeafChild) {
+      ((LeafChild) child).addLeafHook(timeoutHook());
+    }
+  }
+
+  @Override
+  public BlockConfigurable<BlockTimeout> merge(BlockConfigurable<?> other) {
+    // my timeout supersedes any inherited timeout
+
+    return this;
+  }
+
+  private Hook timeoutHook() {
+    return withUnreportedInnerBlock((description, reporting, block) -> {
+      FailOnTimeout.builder().withTimeout(timeout, timeUnit)
+          .build(statementOf(block))
+          .evaluate();
```

Comment:
Essentially this is `JUnit`'s own implementation that I'm using here... it caused a few glitches, hence some reworking of the way that `NotifyingBlock` is used inside tests, since the multi-threaded nature of the above mechanism gave me some `InterruptedException` headaches.